src/cascadia/TerminalApp/CascadiaSettings.cpp

```diff
@@ -452,3 +423,168 @@ void CascadiaSettings::_ValidateAllSchemesExist()
+GUID CascadiaSettings::_GetProfileForArgs(const NewTerminalArgs& newTerminalArgs) const
+{
+    std::optional<int> profileIndex{ std::nullopt };
+    if (newTerminalArgs &&
+        newTerminalArgs.ProfileIndex() != nullptr)
+    {
+        profileIndex = newTerminalArgs.ProfileIndex().Value();
+    }
+    GUID profileGuid = _GetProfileForIndex(profileIndex);
+
+    if (newTerminalArgs)
+    {
+        const auto profileString = newTerminalArgs.Profile();
+
+        // First, try and parse the "profile" argument as a GUID. If it's a
+        // GUID, and the GUID of one of our profiles, then use that as the
+        // profile GUID instead. If it's not, then try looking it up as a
+        // name of a profile. If it's still not that, then just ignore it.
+        if (!profileString.empty())
+        {
+            bool wasGuid = false;
+
+            // Do a quick heuristic check - is the profile 38 chars long (the
+            // length of a GUID string), and does it start with '{'? Because if
+            // it doesn't, it's _definitely_ not a GUID.
+            if (profileString.size() == 38 && profileString[0] == L'{')
+            {
+                try
+                {
+                    const auto newGUID = Utils::GuidFromString(profileString.c_str());
+                    if (FindProfile(newGUID))
+                    {
+                        profileGuid = newGUID;
+                        wasGuid = true;
+                    }
+                }
+                CATCH_LOG();
+            }
+
+            // Here, we were unable to use the profile string as a GUID to
+            // lookup a profile. Instead, try using the string to look the
+            // Profile up by name.
+            if (!wasGuid)
+            {
+                const auto guidFromName = FindGuid(profileString.c_str());
+                if (guidFromName.has_value())
+                {
+                    profileGuid = guidFromName.value();
+                }
+            }
+        }
+    }
+
+    return profileGuid;
+}
+
+// Method Description:
+// - Helper to find the profile GUID for a the profile at the given index in the
+//   list of profiles. If no index is provided, this instead returns the default
+//   profile's guid. This is used by the NewTabProfile<N> ShortcutActions to
+//   create a tab for the Nth profile in the list of profiles.
+// Arguments:
+// - index: if provided, the index in the list of profiles to get the GUID for.
+//   If omitted, instead return the default profile's GUID
+// Return Value:
+// - the Nth profile's GUID, or the default profile's GUID
+GUID CascadiaSettings::_GetProfileForIndex(std::optional<int> index) const
+{
+    GUID profileGuid;
+    if (index)
+    {
+        const auto realIndex = index.value();
+        // If we don't have that many profiles, then do nothing.
+        if (realIndex >= gsl::narrow_cast<decltype(realIndex)>(_profiles.size()))

```

Comment:
Since we're using int and not uint, i think somebody can pass `-1` and beef it down on line 581

on line 581, prefer `.at()` to appease the static analyzer (which is not in effect on these files yet.)